packages/opentelemetry-resources/src/detectors/EnvDetector.ts

```diff
@@ -14,25 +14,142 @@
  * limitations under the License.
  */
 
-import { Detector } from '../types';
+import { Attributes, diag } from '@opentelemetry/api';
+import { getEnv } from '@opentelemetry/core';
+import { SEMRESATTRS_SERVICE_NAME } from '@opentelemetry/semantic-conventions';
 import { ResourceDetectionConfig } from '../config';
-import { IResource } from '../IResource';
-import { envDetectorSync } from './EnvDetectorSync';
+import { ResourceDetector } from '../types';
 
 /**
  * EnvDetector can be used to detect the presence of and create a Resource
  * from the OTEL_RESOURCE_ATTRIBUTES environment variable.
  */
-class EnvDetector implements Detector {
+class EnvDetector implements ResourceDetector {
+  // Type, attribute keys, and attribute values should not exceed 256 characters.
+  private readonly _MAX_LENGTH = 255;
```

Comment:
nit but is there a reason to prefer to have these as `private readonly` fields, as opposed to un-exported module-level constant? the latter seems more "truly" private, easier to type, and more mangle/tree-shaking friendly.

Reply:
no. The implementation was unchanged in this PR but can be fixed in a follow-up. the reason git thinks i changed the impl is because there used to be a different file for the sync detector.